README.md

```diff
@@ -122,6 +122,239 @@ the components involved in the test:
 cd test/integration && dlv test -- -test.run ^TestCrud$
 ``
 
+#### E2E
+
+The Federation-v2 E2E tests can run in an *unmanaged*, *managed*, or *hybrid*
+modes. For both unmanaged and hybrid modes, you will need to bring your own
+clusters. The managed mode runs similarly to integration as it uses the same
+test fixture setup. All of these modes run CRUD operations. CRUD here means
+that the tests will run through each of the requested federated types and
+verify that:
+
+1. the objects are created in the target clusters.
+1. an annotation update is reflected in the objects stored in the target
+   clusters.
+1. a placement update for the object is reflected in the target clusters.
+1. deleted resources are removed from the target clusters.
+
+The read operation is implicit.
+
+##### Managed
+
+The E2E managed tests will spin up the same environment as the
+[Integration](README.md#integration) tests described above and run [CRUD
+(create-read-update-delete)
+checks](https://github.com/kubernetes-sigs/federation-v2/blob/master/test/e2e/crud.go) for
+federated types against that federation. To run:
+
+ - ensure the same binaries are available as described in the
+   [Integration](README.md#integration) section.
+
+To run tests for all types:
+
+```bash
+cd test/e2e
+go test -v
+```
+
+To run tests for a single type:
+
+```bash
+cd test/e2e
+go test -args -v=4 -test.v --ginkgo.focus='"FederatedSecret"'
+```
+
+It may be helpful to use the [delve
+debugger](https://github.com/derekparker/delve) to gain insight into
+the components involved in the test:
+
+```bash
+cd test/e2e
+dlv test -- -v=4 -test.v --ginkgo.focus='"FederatedSecret"'
+```
+
+##### Unmanaged and Hybrid Cluster Setup
+
+The difference between unmanaged and hybrid is that with hybrid, you run
+the federation-v2 controllers in-process as part of executing the test. This
+helps in running a debugger to debug anything in the controllers. On the other
+hand with unmanaged, the controllers are already running in the K8s cluster.
+Both methods require the clusters to have already been joined.
+
+###### Create Clusters
+
+The quickest way to set up clusters for use with unmanaged and hybrid modes is
+to use [minikube](https://kubernetes.io/docs/getting-started-guides/minikube/).
+Once you have minikube installed run:
+
+```bash
+minikube start -p clusterA
+```
+
+###### Deploy the Cluster Registry
+
+Get the latest version of the [Cluster
+Registry](https://github.com/kubernetes/cluster-registry/releases) and run the
+following:
+
+```bash
+crinit aggregated init mycr --host-cluster-context=clusterA
```

Comment:
I ran into an issue with `crinit` when I tried to run it too soon after minikube initialization.  If the storage provisioner hadn't been started before `crinit` was invoked, the etcd pvc would never be bound.  It may be worth documenting this and/or providing a sample check similar to what my deployment script uses:

https://github.com/kubernetes-sigs/federation-v2/pull/57/files#diff-1ed12d3b5689c7a45b1c21e93e23e3d3R53

Reply:
Yes, that can happen. Personally, I think the minikube command should not return until all the components are up and ready, or timeout saying they have not properly come up. That may already be an option.

Reply:
It is not an option afaict.

Reply:
Done.